src/inmanta/server/services/resourceservice.py

```diff
@@ -414,6 +414,109 @@ async def mark_deployed(
             is_increment_notification=True,
         )
 
+    async def _update_deploy_state(
+        self,
+        env: data.Environment,
+        resource_id: ResourceIdStr,
+        timestamp: datetime.datetime,
+        version: int,
+        status: ResourceState,
+        message: str,
+        fail_on_error: bool,
+        is_increment_notification: bool,
+        connection: Optional[Connection] = None,
+    ) -> None:
+        """
+        Factored out the code to set a status on a resource
+
+        This excludes running `mark_done_if_done` and the agent push
+
+        Set the status of the provided resources as deployed
+        :param env: Environment to consider.
+        :param resource_id: resource to mark.
+        :param timestamp: Timestamp for the log message and the resource action entry.
+        :param version: Version of the resources to consider.
+        :param fail_on_error: When encountering an undeployable state: fail or do nothing?.
+        """
+        resource_version_id = resource_id + ",v=" + str(version)
+        logline = LogLine(
+            level=const.LogLevel.INFO,
+            msg=f"Setting {status.value} because of {message}",
+            timestamp=timestamp,
+        )
+
+        assert status in VALID_STATES_ON_STATE_UPDATE
```

Comment:
Perhaps I have to reduce this to just the two states I need now? 

I copied over some functionality for the other states, but it is not really covered by tests.